h/static/scripts/base/upgrade-elements.js

```diff
@@ -13,12 +13,30 @@
  *        order to upgrade it.
  */
 function upgradeElements(root, controllers) {
+  // A helper which replaces the content (including the root element) of
+  // an upgraded element with new markup and re-applies element upgrades to
+  // the new root element
+  function reload(element, html) {
+    if (typeof html !== 'string') {
+      throw new Error('Replacement markup must be a string');
+    }
+    var container = document.createElement('div');
```

Comment:
Rather than assuming that controllers are always `div`s, it would probably be nice to do `element.cloneNode(false)` here instead.